Metro-Records/la-metro-councilmatic
# Implement board member details as snippet

## Overview

This PR adds a Django model for board member headshots and bios and registers as a snippet for editing in the CMS. It also deprecates the `headshot_url` property, instead creating a shared template block that renders the right thing given a person instance and, optionally, a way to request the original aspect ratio, instead of a square thumbnail.

Connects #1176

### Demo

<img width="1512" alt="Screenshot 2024-12-20 at 12 15 58 PM" src="https://github.com/user-attachments/assets/c5dbc5a9-1cea-4335-a02e-e6a014143077" />
<img width="1512" alt="Screenshot 2024-12-20 at 12 16 03 PM" src="https://github.com/user-attachments/assets/16300fe5-f1aa-4d98-ab34-4267d799ab7e" />
<img width="1512" alt="Screenshot 2024-12-20 at 12 16 15 PM" src="https://github.com/user-attachments/assets/490fbaa2-75fe-41f5-adb6-b6e0b150745d" />
<img width="1512" alt="Screenshot 2024-12-20 at 12 16 26 PM" src="https://github.com/user-attachments/assets/8e22c43b-909a-4483-b2a0-f88e19b8882e" />

### Notes

I decided to go ahead and implement this as a snippet, since the `modeladmin` module is deprecated in the next version of Wagtail, so using snippets now makes our migration path easier moving forward. I'm wondering if we might want to do this in #1185, too. https://docs.wagtail.org/en/v5.2.1/reference/contrib/modeladmin/migrating_to_snippets.html

## Testing Instructions

- Log into the review app, then check out the Board Member Details section of the CMS.
- Try some filters and verify they work as expected.
- Try editing some headshots and bios. Confirm the previews render correctly.
- Publish some of your edits. Confirm the changes are visible on the live site.
